src/solvers/dg/dg_2d.jl

```diff
@@ -363,61 +367,188 @@ function calc_volume_integral!(du::AbstractArray{<:Any,4}, u, nonconservative_te
     split_form_kernel!(du, u, nonconservative_terms, equations, volume_flux_dg, dg, cache, element, 1 - alpha_element)
 
     # Calculate FV volume integral contribution
-    fv_kernel!(du, u, equations, volume_flux_fv, dg, cache, element, alpha_element)
+    fv_kernel!(du, u, nonconservative_terms, equations, volume_flux_fv, dg, cache, element, alpha_element)
   end
 
   return nothing
 end
 
-@inline function fv_kernel!(du::AbstractArray{<:Any,4}, u::AbstractArray{<:Any,4},
+@inline function fv_kernel!(du::AbstractArray{<:Any,4}, u::AbstractArray{<:Any,4}, nonconservative_terms,
                             equations, volume_flux_fv, dg::DGSEM, cache, element, alpha=true)
-  @unpack fstar1_threaded, fstar2_threaded = cache
+  @unpack fstar1_L_threaded, fstar1_R_threaded, fstar2_L_threaded, fstar2_R_threaded = cache
   @unpack inverse_weights = dg.basis
 
   # Calculate FV two-point fluxes
-  fstar1 = fstar1_threaded[Threads.threadid()]
-  fstar2 = fstar2_threaded[Threads.threadid()]
-  calcflux_fv!(fstar1, fstar2, u, equations, volume_flux_fv, dg, element)
+  fstar1_L = fstar1_L_threaded[Threads.threadid()]
+  fstar2_L = fstar2_L_threaded[Threads.threadid()]
+  fstar1_R = fstar1_R_threaded[Threads.threadid()]
+  fstar2_R = fstar2_R_threaded[Threads.threadid()]
+  calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u,
+               nonconservative_terms, equations, volume_flux_fv, dg, element)
 
   # Calculate FV volume integral contribution
   for j in eachnode(dg), i in eachnode(dg)
     for v in eachvariable(equations)
       du[v, i, j, element] += ( alpha *
-                                (inverse_weights[i] * (fstar1[v, i+1, j] - fstar1[v, i, j]) +
-                                 inverse_weights[j] * (fstar2[v, i, j+1] - fstar2[v, i, j])) )
-
+                                (inverse_weights[i] * (fstar1_L[v, i+1, j] - fstar1_R[v, i, j]) +
+                                 inverse_weights[j] * (fstar2_L[v, i, j+1] - fstar2_R[v, i, j])) )
     end
   end
 
   return nothing
 end
 
-@inline function calcflux_fv!(fstar1, fstar2, u::AbstractArray{<:Any,4},
-                              equations, volume_flux_fv, dg::DGSEM, element)
 
-  fstar1[:, 1,            :] .= zero(eltype(fstar1))
-  fstar1[:, nnodes(dg)+1, :] .= zero(eltype(fstar1))
+"""
+    calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u_leftright,
+                 nonconservative_terms::Val{false}, equations, volume_flux_fv, dg, element_id)
+
+Calculate the finite volume fluxes inside the elements (without non-conservative terms).
```

Comment:
```suggestion
Calculate the finite volume fluxes inside the elements (**without non-conservative terms**).
```

Same reason as for Taam above